test/remove.js

```diff
@@ -232,42 +274,38 @@ test('.where() multiple clause', function (assert) {
     ],
     copy = slice.call(records),
 
-    result = list(records).where({
-      billingEmail: 'dennis@example.com',
-      lineItems: {
-        sku: 'h617xrh'
-      }
-    }),
+    query = {
+      "lastName": "Chambers",
+    },
 
-    expected = [
-      {
-        "id": "ci6r6aliv00007poxc2zgnjvf",
-        "date": "2014-12-30 05:29:28",
-        "billingEmail": "dennis@example.com",
-        "firstName": "Dennis",
-        "lastName": "Chambers",
-        "lineItems": {
-          "name": "Zildjian K Custom Organic Ride - 21\"",
-          "sku": "h617xrh",
-          "quantity": "1",
-          "total": "379.95"
-        }
-      }
-    ];
+    result = list({
+      list: records,
+      async: true
+    });
 
-  assert.deepEqual(result, expected,
-    'should contain all the expected records.');
+  result = result.removeWhere(query);
+  result.subscribe(
+    function onNext(newCollection) {
```

Comment:
oops.. I didn't catch this before, but onNext is supposed to handle a single index at a time rather than the entire collection. That's what makes it compatible with infinite streaming.

In other words, we should really be delegating these methods to an underlying RxObservable which is the actual representation of the collection for async operations. Make sense?